frameworks/MLNet/exec.py

```diff
@@ -0,0 +1,116 @@
+import logging
+import math
+import os
+import pandas as pd
+from amlb.benchmark import TaskConfig
+from amlb.data import Dataset
+from amlb.datautils import reorder_dataset
+from amlb.results import NoResultError, save_predictions
+from amlb.utils import dir_of, path_from_split, run_cmd, split_path, Timer
+import json
+from frameworks.shared.callee import call_run, result, save_metadata
+
+log = logging.getLogger(__name__)
+
+def run(dataset: Dataset, config: TaskConfig):
+    avaible_task_list = ['classification', 'regression']
+    if config.type not in avaible_task_list:
+        raise ValueError('{} is not supported.'.format(config.type))
+
+    dir_path = os.path.dirname(os.path.realpath(__file__))
+    DOTNET_INSTALL_DIR = os.path.join(dir_path, '.dotnet')
+    os.environ['MODELBUILDER_AUTOML'] = 'NNI'
+    os.environ['DOTNET_ROOT'] = DOTNET_INSTALL_DIR
+    mlnet = os.path.join(DOTNET_INSTALL_DIR, 'mlnet')
+    save_metadata(config)
+    name = config.name
+    temp_output_folder = os.path.join(config.output_dir,name)
+    if not os.path.exists(temp_output_folder):
+        os.mkdir(temp_output_folder)
+    train_time_in_seconds = config.max_runtime_seconds
+    log_path = os.path.join(temp_output_folder, 'log.txt')
+    sub_command = config.type
+    column_num = dataset.train.X.shape[1]
+    columns=['column_{}'.format(i) for i in range(column_num)]
+    train_df = pd.DataFrame(dataset.train.X, columns=columns)
+    train_df['label'] = dataset.train.y
+    test_df = pd.DataFrame(dataset.test.X, columns=columns)
+    test_df['label'] = dataset.test.y
+    train_dataset = os.path.join(temp_output_folder, 'train.csv')
+    test_dataset = os.path.join(temp_output_folder, 'test.csv')
+    log.info("saving train to {}".format(train_dataset))
+    train_df.to_csv(train_dataset, index=False, header=True)
+    test_df.to_csv(test_dataset, index=False, header=True)
+    with Timer() as training:
+        cmd = '{} {}'.format(mlnet, sub_command)
```

Comment:
feel free to use f-strings everywhere if you think it improves readability, also consecutive strings are automatically concatenated in Python:
```python
cmd = f"{mlnet} {sub_command}"\
      f" -- dataset {train_dataset} --test-dataset {test_dataset}"
```
or you can use a single multiline:
```python
cmd = f"""{mlnet} {sub_command}
--dataset {train_dataset} --test-dataset {test_dataset}
--train-time {train_time_in_seconds}
"""
```

Reply:
### Resolved